src/sql/src/catalog.rs

```diff
@@ -1126,6 +1134,8 @@ pub enum CatalogError {
     ConfigAlreadyExists(String),
     /// Builtin migrations failed.
     FailedBuiltinSchemaMigration(String),
+    /// Catalog state missing information
+    MissingState(String),
```

Comment:
I think you should be able to avoid this by not storing the AWS info in the connection config; see future comment.